web_src/js/features/comp/EditorUpload.ts

```diff
@@ -8,43 +8,45 @@ import {
   generateMarkdownLinkForAttachment,
 } from '../dropzone.ts';
 import type CodeMirror from 'codemirror';
+import type EasyMDE from 'easymde';
 
 let uploadIdCounter = 0;
 
 export const EventUploadStateChanged = 'ce-upload-state-changed';
 
-export function triggerUploadStateChanged(target) {
+export function triggerUploadStateChanged(target: HTMLElement) {
   target.dispatchEvent(new CustomEvent(EventUploadStateChanged, {bubbles: true}));
 }
 
-function uploadFile(dropzoneEl, file) {
+function uploadFile(dropzoneEl: HTMLElement, file: File) {
   return new Promise((resolve) => {
     const curUploadId = uploadIdCounter++;
-    file._giteaUploadId = curUploadId;
+    (file as any)._giteaUploadId = curUploadId;
     const dropzoneInst = dropzoneEl.dropzone;
-    const onUploadDone = ({file}) => {
+    const onUploadDone = ({file}: {file: any}) => {
       if (file._giteaUploadId === curUploadId) {
         dropzoneInst.off(DropzoneCustomEventUploadDone, onUploadDone);
         resolve(file);
       }
     };
     dropzoneInst.on(DropzoneCustomEventUploadDone, onUploadDone);
+    // @ts-expect-error - handleFiles not present?
     dropzoneInst.handleFiles([file]);
```

Comment:
It does exist

![image](https://github.com/user-attachments/assets/fb39dc59-a98b-4a99-a34f-3b3d9333af07)

Reply:
Maybe something wrong with our Dropzone type.

Reply:
`tsc` definitely reports this:

```
web_src/js/features/comp/EditorUpload.ts:33:18 - error TS2339: Property 'handleFiles' does not exist on type 'Dropzone'.

33     dropzoneInst.handleFiles([file]);
                    ~~~~~~~~~~~
```

Reply:
It's a bug in `@types/dropzone`:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/7ee567329313108f5a8895a6dc6eafd75cf64546/types/dropzone/index.d.ts#L162

Will update the comment.

Reply:
Actually, as seen in the second link above, `handleFile` just loops the array and calls `addFile`, so I changed the code to just use `addFile` which is typed. Still it's not entirely correct because `addFile` wants a `DropzoneFile`, not a `File` but that is an existing bug. I've worked around by casting the type and adding a comment.

Reply:
Yup, the Dropzone has been abused for long time ..... I managed to get some abuses fixed but there are still some problems. I guess we can tolerate them until there is a strong requirement to rewrite more code and do full a test.

Reply:
I'd rather like to replace Dropzone with a simple list of files and some buttons besides them. That list should not show images that are already embedded in the post, only when editing, show these so they can be deleted.